modules/gmc-agent/src/main/java/org/gridgain/dto/action/query/ScanQueryArgument.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gridgain.dto.action.query;
+
+import org.apache.ignite.internal.util.typedef.internal.S;
+
+import java.util.UUID;
+
+/**
+ * DTO fro scan query argument.
+ */
+public class ScanQueryArgument {
+    /** Query ID. */
+    private String qryId;
+
+    /** Cache name for query. */
+    private String cacheName;
+
+    /** Filter text. */
+    private String filter;
```

Comment:
I do not like this filter. Lets implement something like HZ have?
Like a set of filter rules. For example (just to show idea)
`year > 2010 & salary > 300 and position=manager`

Reply:
Create the ticket, add todo.